javaslang/src/test/java/javaslang/collection/euler/Euler43Test.java

```diff
@@ -54,25 +54,22 @@ public void shouldSolveProblem43() {
     private static Seq<Long> tenDigitPandigitalsWithProperty() {
         final CharSeq ALL_DIGITS = CharSeq.of("0123456789");
         final List<Integer> DIVISORS = List.of(2, 3, 5, 7, 11, 13, 17);
-        final char DUMMY_CHAR = Character.MIN_VALUE;
 
         return ALL_DIGITS
                 .combinations(2)
                 .flatMap(CharSeq::permutations)
-                .map(cs -> Tuple.of(cs, cs.prepend(DUMMY_CHAR), Integer.valueOf(cs.mkString())))
                 .flatMap(firstTwoDigits -> DIVISORS
                         .foldLeft(List.of(firstTwoDigits), (accumulator, divisor) -> accumulator
                                 .flatMap(digitsSoFar -> ALL_DIGITS
-                                        .removeAll(digitsSoFar._1)
-                                        .map(nextDigit -> Tuple.of(digitsSoFar._1.append(nextDigit), digitsSoFar._2.tail().append(nextDigit)))
-                                        .map(digitsToTest -> Tuple.of(digitsToTest._1, digitsToTest._2, Integer.valueOf(digitsToTest._2.mkString())))
+                                        .removeAll(digitsSoFar)
+                                        .map(nextDigit -> digitsSoFar.append(nextDigit))
```

Comment:
Yes, but I think this a bit more verbose version naming what is being
appended (nextDigit) helps understanding/readability. If shortened to a
method reference the whole concept of "nextdigit" is implicit.

On 22 August 2016 at 01:19, Daniel Dietrich <notifications@github.com>
wrote:

> In javaslang/src/test/java/javaslang/collection/euler/Euler43Test.java
> <https://github.com/javaslang/javaslang/pull/1514#discussion_r75605623>:
>
> >                  .flatMap(firstTwoDigits -> DIVISORS
> >                          .foldLeft(List.of(firstTwoDigits), (accumulator, divisor) -> accumulator
> >                                  .flatMap(digitsSoFar -> ALL_DIGITS
> > -                                        .removeAll(digitsSoFar._1)
> > -                                        .map(nextDigit -> Tuple.of(digitsSoFar._1.append(nextDigit), digitsSoFar._2.tail().append(nextDigit)))
> > -                                        .map(digitsToTest -> Tuple.of(digitsToTest._1, digitsToTest._2, Integer.valueOf(digitsToTest._2.mkString())))
> > +                                        .removeAll(digitsSoFar)
> > +                                        .map(nextDigit -> digitsSoFar.append(nextDigit))
>
> here we can use the shorted method reference: .map(digitsSoFar::append)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/javaslang/javaslang/pull/1514/files/3e3032fae50687edf43e07a348c26d709b0c77d6#r75605623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADRVwLKbUo-dgmUab3x_oziCE3Y4nGgyks5qiNz5gaJpZM4JpO4g>
> .
>